### PR TITLE
Unifying the docs for the Poll class.

### PR DESCRIPTION
**Pull request by RobiMez**
### Checklist for PRs

- [x] Added `.. versionadded:: version`, `.. versionchanged:: version` or `.. deprecated:: version` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
- [ ] Created new or adapted existing unit tests
- [ ] Documented code changes according to the [CSI standard](https://standards.mousepawmedia.com/en/stable/csi.html)
- [ ] Added myself alphabetically to `AUTHORS.rst` (optional)
- [ ] Added new classes & modules to the docs and all suitable `__all__` s

### CODE REVIEW

File: telegram/_bot.py [hunk 1]
```diff
@@ -5951,8 +5951,9 @@ async def send_poll(
                 :tg-const:`telegram.Poll.REGULAR`, defaults to :tg-const:`telegram.Poll.REGULAR`.
             allows_multiple_answers (:obj:`bool`, optional): :obj:`True`, if the poll allows
                 multiple answers, ignored for polls in quiz mode, defaults to :obj:`False`.
-            correct_option_id (:obj:`int`, optional): 0-based identifier of the correct answer
-                option, required for polls in quiz mode.
+            correct_option_id (:obj:`int`): Optional. A zero based identifier of the correct answer
+                option. Available only for closed polls in the quiz mode, which were sent (not 
+                forwarded), by the bot or to a private chat with the bot.
```
Comment (harshil21) [inline telegram/_bot.py:5956]: Bot methods should stick to official documentation from core.telegram.org/bots/api

Please revert

File: telegram/_poll.py [hunk 2]
```diff
@@ -126,13 +126,18 @@ class Poll(TelegramObject):
...
-            Available only for polls in the quiz mode, which are closed, or was sent (not
-            forwarded) by the bot or to the private chat with the bot.
+        correct_option_id (:obj:`int`, optional)): A zero based identifier of the correct answer option.
+            Available only for closed polls in the quiz mode, which were sent (not
+            forwarded), by the bot or to a private chat with the bot.
         explanation (:obj:`str`, optional): Text that is shown when a user chooses an incorrect
-            answer or taps on the lamp icon in a quiz-style poll, 0-200 characters.
-        explanation_entities (List[:class:`telegram.MessageEntity`], optional): Special entities
+            answer or taps on the lamp icon in a quiz-style poll, 0-200 characters with at most
+            2 line feeds after entities parsing.
```
Comment (harshil21) [inline telegram/_poll.py:134]: Please remove the "2 line feeds..." senetence

File: telegram/_poll.py [hunk 3]
```diff
@@ -147,19 +152,27 @@ class Poll(TelegramObject):
         is_anonymous (:obj:`bool`): :obj:`True`, if the poll is anonymous.
         type (:obj:`str`): Poll type, currently can be :attr:`REGULAR` or :attr:`QUIZ`.
         allows_multiple_answers (:obj:`bool`): :obj:`True`, if the poll allows multiple answers.
-        correct_option_id (:obj:`int`): Optional. Identifier of the correct answer option.
-        explanation (:obj:`str`): Optional. Text that is shown when a user chooses an incorrect
-            answer or taps on the lamp icon in a quiz-style poll.
+        correct_option_id (:obj:`int`): Optional. A zero based identifier of the correct answer option.
+            Available only for closed polls in the quiz mode, which were sent (not
+            forwarded), by the bot or to a private chat with the bot.
+        explanation (:obj:`str`, optional): Text that is shown when a user chooses an incorrect
```
Comment (harshil21) [inline telegram/_poll.py:158]:
```suggestion
        explanation (:obj:`str`): Optional. Text that is shown when a user chooses an incorrect
```

File: telegram/_poll.py [hunk 4]
```diff
@@ -147,19 +152,27 @@ class Poll(TelegramObject):
...
-            automatically closed.
+        open_period (:obj:`int`, optional): Amount of time in seconds the poll will be active
+            after creation, 5-600. Can't be used together with :paramref:`close_date`.
+        close_date (:obj:`int` | :obj:`datetime.datetime`, optional): Point in time (Unix
+            timestamp) when the poll will be automatically closed. Must be at least 5 and no
+            more than 600 seconds in the future. Can't be used together with
+            :paramref:`open_period`.
+            For timezone naive :obj:`datetime.datetime` objects, the default timezone of the
+            bot will be used, which is UTC unless :attr:`telegram.ext.Defaults.tzinfo` is
+            used.
```
Comment (harshil21) [inline telegram/_poll.py:175]: This should not be unified with `send_poll` docs. Please revert

**Review by harshil21 — CHANGES_REQUESTED** [review]
Thanks for the PR, left some comments below

File: telegram/_bot.py (hunk 1 above)
Comment (RobiMez) [reply telegram/_bot.py]: done

File: telegram/_poll.py (hunk 2 above)
Comment (RobiMez) [reply telegram/_poll.py]: done

File: telegram/_poll.py (hunk 3 above)
Comment (RobiMez) [reply telegram/_poll.py]: done

File: telegram/_poll.py (hunk 4 above)
Comment (RobiMez) [reply telegram/_poll.py]: done

File: telegram/_poll.py [hunk 5]
```diff
@@ -126,13 +126,17 @@ class Poll(TelegramObject):
...
-        correct_option_id (:obj:`int`, optional): 0-based identifier of the correct answer option.
-            Available only for polls in the quiz mode, which are closed, or was sent (not
-            forwarded) by the bot or to the private chat with the bot.
+        correct_option_id (:obj:`int`, optional)): A zero based identifier of the correct answer option.
+            Available only for closed polls in the quiz mode, which were sent (not
+            forwarded), by the bot or to a private chat with the bot.
         explanation (:obj:`str`, optional): Text that is shown when a user chooses an incorrect
             answer or taps on the lamp icon in a quiz-style poll, 0-200 characters.
-        explanation_entities (List[:class:`telegram.MessageEntity`], optional): Special entities
+        explanation_entities (List[:class:`telegram.MessageEntity`]): Special entities
```
Comment (harshil21) [inline telegram/_poll.py:134]:
```suggestion
        explanation_entities (List[:class:`telegram.MessageEntity`], optional): Special entities
```

File: telegram/_poll.py [hunk 6]
```diff
@@ -147,9 +151,12 @@ class Poll(TelegramObject):
...
         type (:obj:`str`): Poll type, currently can be :attr:`REGULAR` or :attr:`QUIZ`.
         allows_multiple_answers (:obj:`bool`): :obj:`True`, if the poll allows multiple answers.
-        correct_option_id (:obj:`int`): Optional. Identifier of the correct answer option.
+        correct_option_id (:obj:`int`): Optional. A zero based identifier of the correct answer option.
+            Available only for closed polls in the quiz mode, which were sent (not
+            forwarded), by the bot or to a private chat with the bot.
         explanation (:obj:`str`): Optional. Text that is shown when a user chooses an incorrect
-            answer or taps on the lamp icon in a quiz-style poll.
+            answer or taps on the lamp icon in a quiz-style poll, 0-200 characters with at most
+            2 line feeds after entities parsing.
```
Comment (harshil21) [inline telegram/_poll.py:159]: remove the "2 line feeds" here as well

**Review by harshil21 — COMMENTED** [review]
2 more missed ones-

File: telegram/_poll.py [hunk 7]
```diff
@@ -126,13 +126,17 @@ class Poll(TelegramObject):
         is_anonymous (:obj:`bool`): :obj:`True`, if the poll is anonymous.
         type (:obj:`str`): Poll type, currently can be :attr:`REGULAR` or :attr:`QUIZ`.
         allows_multiple_answers (:obj:`bool`): :obj:`True`, if the poll allows multiple answers.
-        correct_option_id (:obj:`int`, optional): 0-based identifier of the correct answer option.
-            Available only for polls in the quiz mode, which are closed, or was sent (not
-            forwarded) by the bot or to the private chat with the bot.
+        correct_option_id (:obj:`int`, optional)): A zero based identifier of the correct answer option.
```
Comment (harshil21) [inline telegram/_poll.py:129]:
```suggestion
        correct_option_id (:obj:`int`, optional): A zero based identifier of the correct answer option.
```
extra bracket

**Review by harshil21 — APPROVED** [review]

File: telegram/_poll.py (hunk 6 above)
Comment (RobiMez) [reply telegram/_poll.py]: done

**Comment by harshil21** [comment]
pre-commit is failing since the line is too long (L129 & 154 in _poll.py). It should be < 100 chars

**Comment by RobiMez** [comment]
ill add some breaks

**Comment by RobiMez** [comment]
done , also a few questions , was the line feed bit unnecessary ? is it because its a niche thing that people don't run into or it just didnt belong there ?

**Comment by RobiMez** [comment]
oh wait , trailing whitespace

**Comment by harshil21** [comment]
@RobiMez it&#x27;s the latter, the official documentation of <a href="https://core.telegram.org/bots/api#poll">Poll </a>doesn&#x27;t list that, so we don&#x27;t write it.

**Comment by RobiMez** [comment]
@harshil21 Ohhh , makes sense , getting the hang of sphinx and your linting checks too. Ill tackle another class tomorrow as its getting late here. any priorities / which one needs most work ?

**Comment by harshil21** [comment]
> @harshil21 any priorities / which one needs most work?

no priorities as of now, feel free to dive in the code and pick any class(es) you find which has a large discrepancy. Try to do multiple classes together so as to not produce many PRs for the same thing.

**Pull request merged**